src/clipboard_handlers/data_validation_clipboard.ts

```diff
@@ -84,40 +95,43 @@ export class DataValidationClipboardHandler extends AbstractCellClipboardHandler
   }
 
   private pasteDataValidation(
-    origin: CellPosition,
+    origin: Maybe<ClipboardDataValidationRule>,
     target: CellPosition,
     isCutOperation?: boolean
   ) {
-    const rule = this.getters.getValidationRuleForCell(origin);
-    if (!rule) {
-      const targetRule = this.getters.getValidationRuleForCell(target);
-      if (targetRule) {
-        // Remove the data validation rule on the target cell
-        this.adaptDataValidationRule(target.sheetId, targetRule, [], [positionToZone(target)]);
-      }
-      return;
-    }
-    const zone = positionToZone(target);
-    for (const range of rule.ranges) {
-      if (isInside(origin.col, origin.row, range.zone)) {
-        const toRemoveZone: Zone[] = [];
-        if (isCutOperation) {
-          toRemoveZone.push(positionToZone(origin));
+    if (origin) {
+      const zone = positionToZone(target);
+      const rule = this.getters.getValidationRuleForCell(origin.position) || origin.rule;
+      if (!rule) {
+        const targetRule = this.getters.getValidationRuleForCell(target);
+        if (targetRule) {
+          // Remove the data validation rule on the target cell
+          this.adaptDataValidationRule(target.sheetId, targetRule, [], [zone]);
         }
-        if (origin.sheetId === target.sheetId) {
-          this.adaptDataValidationRule(origin.sheetId, rule, [zone], toRemoveZone);
-        } else {
-          this.adaptDataValidationRule(origin.sheetId, rule, [], toRemoveZone);
-          const copyToRule = this.getDataValidationRuleToCopyTo(target.sheetId, rule);
-          this.adaptDataValidationRule(target.sheetId, copyToRule, [zone], []);
+        return;
+      }
+      const toRemoveZone: Zone[] = [];
+      if (isCutOperation) {
+        toRemoveZone.push(positionToZone(origin.position));
+      }
+      if (origin.position.sheetId === target.sheetId) {
+        const copyToRule = this.getDataValidationRuleToCopyTo(target.sheetId, rule, false);
+        this.adaptDataValidationRule(origin.position.sheetId, copyToRule, [zone], toRemoveZone);
+      } else {
+        /** If the origin sheet was deleted before pasting the data, no need to adapt the DV rules  */
+        if (this.getters.getSheetIds().includes(origin.position.sheetId)) {
+          this.adaptDataValidationRule(origin.position.sheetId, rule, [], toRemoveZone);
```

Comment:
the zone should be removed from the original rule, not the one in the clipboard.
```suggestion
        const originRule = this.getters.getValidationRuleForCell(origin.position)
        if (originRule) {
          this.adaptDataValidationRule(origin.position.sheetId, originRule, [], toRemoveZone);
```
And it allows to remove `this.getters.getValidationRuleForCell(origin.position) ||` above